app/controllers/application_controller.rb

```diff
@@ -16,6 +18,16 @@ def routing_error!
     raise ActionController::RoutingError, %|(No route matches [GET] "#{request.path}")|
   end
 
+  def validate_page_params(page_param, total_pages)
+    return if page_param.nil?
+    page_num = page_param.to_i
+    raise PageNotFound if page_num > total_pages || page_num <= 0 || float_value?(page_param)
+  end
+
+  def float_value?(page_param)
```

Comment:
@fukumame 
We can ignore these edge case.
Because it is too hard to case sensitively :<

``` ruby
"1.hoge".to_f == "1".to_i
#=> true
```